src/include/Token.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <cstdint>
+#include <string>
+
+enum struct TokenType {
+    LeftParen, RightParen, LeftBrace, RightBrace,
```

Comment:
These look grouped - maybe use comments to delimit different "groups" of tokens (`delimiters`, `operators`, etc.) for the sake of readability?

Reply:
Haha, I actually had that originally but removed it because it looked a bit odd to me. I can add it back though.

Reply:
I think it shouldn't look too weird 👍 

```cpp
enum struct TokenType {
   // ...something about a group
   LeftParen, RightParen, LeftBrace, RightBrace,
   
   // ...next group
   bla, bla, bla

}
```

Reply:
I think it's the extra space between the groups that makes it work---it looks really dense otherwise. Anyway, updated.